src/core/smtp/MCSMTPSession.cpp

```diff
@@ -672,17 +672,22 @@ void SMTPSession::sendMessage(Address * from, Array * recipients, Data * message
         goto err;
     }
     else if (r != MAILSMTP_NO_ERROR) {
-        if ((responseCode == 550) && (response != NULL) && (response->hasPrefix(MCSTR("5.3.4")))) {
-            * pError = ErrorNeedsConnectToWebmail;
-            goto err;
-        }
-        else {
-            * pError = ErrorSendMessage;
-            MC_SAFE_REPLACE_COPY(String, mLastSMTPResponse, response);
-            mLastLibetpanError = r;
-            mLastSMTPResponseCode = responseCode;
-            goto err;
+        if ((responseCode == 550) && (response != NULL)) {
+            if (response->hasPrefix(MCSTR("5.3.4"))) {
```

Comment:
Hello Hoà,

1. Could you remove everything that's after "5.7.1 "?

* "5.7.1 " by itself is too common case to consider it permission-related.
It can also be concerned with SMTP relay issues.
I can check, for example, like this:

if (response->hasPrefix(MCSTR("5.7.1 ")) &&
strstr(response->lowercaseString()->UTF8String(), "permission") !=
NULL)


2. Could you take the opportunity to improve that line and add a space
after "5.3.4"

* For sure

On Fri, Aug 7, 2015 at 6:23 PM, Hoà V. DINH <notifications@github.com>
wrote:

> In src/core/smtp/MCSMTPSession.cpp
> <https://github.com/MailCore/mailcore2/pull/1203#discussion_r36529663>:
>
> > @@ -672,17 +672,22 @@ void SMTPSession::sendMessage(Address * from, Array * recipients, Data * message
> >          goto err;
> >      }
> >      else if (r != MAILSMTP_NO_ERROR) {
> > -        if ((responseCode == 550) && (response != NULL) && (response->hasPrefix(MCSTR("5.3.4")))) {
> > -            * pError = ErrorNeedsConnectToWebmail;
> > -            goto err;
> > -        }
> > -        else {
> > -            * pError = ErrorSendMessage;
> > -            MC_SAFE_REPLACE_COPY(String, mLastSMTPResponse, response);
> > -            mLastLibetpanError = r;
> > -            mLastSMTPResponseCode = responseCode;
> > -            goto err;
> > +        if ((responseCode == 550) && (response != NULL)) {
> > +            if (response->hasPrefix(MCSTR("5.3.4"))) {
>
> Could you take the opportunity to improve that line and add a space after
> "5.3.4"? Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MailCore/mailcore2/pull/1203/files#r36529663>.
>